SemanticUI/Modules/Dropdown/Select.elm

```diff
@@ -0,0 +1,230 @@
+        (\makeNoSelect -> makeNoSelect div [ class "text default" ] [ text "None Selected" ])
+        (\makeSelect _ a -> makeSelect div [ class "text" ] [ text (toString a) ])
+
+Example of inline select :
+
+    Select.select
+        (Select.init { identifier = "inline1", onToggle =  ToggleInline1, onSelect = SelectInline1, choices = [ "today", "this week", "this month" ]} |> Select.inline True)
+        model.inline1
+        model.inline1Selection
+        div
+        []
+        (\makeChoice isList c ->
+            makeChoice div
+                [ classList [ ( "active", isList && model.inline1Selection == c ) ] ]
+                [ text c ]
+        )
+
+-}
+
+import SemanticUI.Modules.Common exposing (Render, WrappedNode)
+import SemanticUI.Modules.Dropdown as Dropdown exposing (dropdown)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+
+
+{-| Configuration for a select
+
+  - identifier - unique identifier
+  - onToggle - message to handle dropdown state changes
+  - toggleEvent - what casuse dropdown state change (default OnClick)
+  - readOnly - is the selection read only or not
+  - onSelect - messages to handle selection change (when clicking on choice)
+  - choices - the list of possible choices
+  - inline - should the selection be inline or not
+    (dropdown does not have "selection" class but "inline" and
+    "text" class is added to the unexpanded item)
+
+-}
+type alias Config r c s msg =
+    { r
+        | identifier : String
+        , onToggle : Dropdown.State -> msg
+        , toggleEvent : Dropdown.ToggleEvent
+        , readOnly : Bool
+        , onSelect : s -> msg
+        , choices : List c
+        , inline : Bool
+        , makeSelectToggle : WrappedNode msg -> Html msg
+    }
+
+
+{-| Create a default Config given, unique identifier, dropdown state change handler
+selection handler and list of choices
+-}
+init :
+    { r
+        | identifier : String
+        , onToggle : Dropdown.State -> msg
+        , onSelect : s -> msg
+        , choices : List c
+    }
+    -> Config {} c s msg
+init { identifier, onToggle, onSelect, choices } =
+    { identifier = identifier
+    , onToggle = onToggle
+    , toggleEvent = Dropdown.OnClick
+    , readOnly = False
+    , onSelect = onSelect
+    , choices = choices
+    , inline = False
+    , makeSelectToggle = \makeSelect -> makeSelect i [ class "dropdown icon" ] []
+    }
+
+{-| Modify a Config's makeSelectToggle value
+-}
+makeSelectToggle :
+    (WrappedNode msg -> Html msg)
+    -> { a | makeSelectToggle : WrappedNode msg -> Html msg }
+    -> { a | makeSelectToggle : WrappedNode msg -> Html msg }
+makeSelectToggle b a =
+    { a | makeSelectToggle = b }
+
+{-| Modify a Config's inline value
+-}
+inline : Bool -> { a | inline : Bool } -> { a | inline : Bool }
+inline b a =
+    { a | inline = b }
+
+
+type alias State r a =
+    { r
+        | dropdownState : Dropdown.State
+        , selectedValue : a
+    }
```

Comment:
I suppose Checkbox also uses this extended style for State

```
checkbox : Config msg -> { r | checked : Bool } -> Html msg
```

...but I personally just end up using `State {} (Maybe String)`. E.g.

```elm
dropdown1State : Select.State {} (Maybe String)
dropdown2State : Select.State {} Int
dropdown3State : Select.State {} X
```

I suppose this style could be somewhat useful if one wants to extend the dropdown module with extra generic functionality. I feel like I probably don't mind if the arguments to the functions are anonymous types in this style, but do we want the exposed types to carry around this `r` parameter?